core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -167,8 +171,26 @@ class ControllerContext {
       .getOrElse(topicPartition.partition, ReplicaAssignment.empty)
   }
 
+  /**
+   * This method should be called whenever there is an update to the partitionAssignments map
```

Comment:
Nit: that's a little misleading--if it were called that frequently, the lazilyUpdatedAssignmentsByBroker field wouldn't be lazy. What you really mean is "before any access to lazilyUpdatedAssignmentsByBroker", i.e., this is the mechanism by which it's lazily updated, but it should not be called otherwise.

Reply:
Good catch. The comment is obsolete, and I've updated it.